Modules/_ssl.c

```diff
@@ -6241,6 +6338,66 @@ _ssl_get_default_verify_paths_impl(PyObject *module)
     return NULL;
 }
 
+/*[clinic input]
+@critical_section
+_ssl.get_sigalgs
+[clinic start generated code]*/
+
+static PyObject *
+_ssl_get_sigalgs_impl(PyObject *module)
+/*[clinic end generated code: output=ab0791b63856854b input=bf74cdad3a19d29e]*/
+{
+#if OPENSSL_VERSION_NUMBER >= 0x30400000L
+    const char *sigalgs_list, *sigalg, *end;
+    PyObject *item, *result = NULL;
+    size_t len;
+
+    if ((sigalgs_list = SSL_get1_builtin_sigalgs(NULL)) == NULL) {
+        PyErr_NoMemory();
+        goto error;
+    }
+
+    result = PyList_New(0);
+    if (result == NULL) {
+        PyErr_NoMemory();
+        goto error;
+    }
+
+    sigalg = sigalgs_list;
+    while (sigalg) {
+        end = strchr(sigalg, ':');
+        len = end? end - sigalg : strlen(sigalg);
+
+        // Alg names are plain ASCII, so there's no chance of a decoding
+        // error here. However, an allocation failure could occur when
+        // constructing the Unicode version of the names.
+        item = PyUnicode_DecodeASCII(sigalg, len, "strict");
+        if (item == NULL) {
+            goto error;
+        }
+
+        if (PyList_Append(result, item) == -1) {
+            Py_DECREF(item);
+            goto error;
+        }
+
+        Py_DECREF(item);
+        sigalg = end? end + 1 : end;
+    }
```

Comment:
All of this can be simplified to:

```c
result = PyUnicode_FromStringAndSize(sigalgs_list, strlen(sigalgs_list));
OPENSSL_free((void *)sigalgs_list);
return PyUnicode_Split(result, _Py_LATIN1_CHR(':'), -1);
```